pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -332,7 +337,7 @@ def _track_callback_metrics(self, eval_results):
                     flat['checkpoint_on'] = flat['val_loss']
                     flat['early_stop_on'] = flat['val_loss']
                 self.trainer.logger_connector.callback_metrics.update(flat)
-                if self.trainer.testing:
+                if self.trainer.state in (TrainerState.TESTING, TrainerState.VALIDATING):
```

Comment:
```suggestion
                if self.trainer.evaluating:
```

Reply:
That is not the same. `trainer.evaluating` checks for `RunnnigStage.VALIDATING or RunnnigStage.TESTING` not `TrainerState.VALIDATING or TrainerState.TESTING`